online_adaptation/replay_buffers.py

```diff
@@ -0,0 +1,48 @@
+import random
+import collections
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+base = torch.tensor([2 ** i for i in reversed(range(5))], dtype=torch.float32)
+
+
+class ReplayBuffer(Dataset):
+    def __init__(self, max_size, num_classes):
+        self.buffers = {k: collections.deque(maxlen=max_size) for k in range(num_classes)}
+
+    def add(self, observation: torch.Tensor, action: torch.Tensor):
+        assert (action.long() == action).all(), "Actions must be integers"
+        targets = action.argwhere().flatten().tolist()
+        for target in targets:
+            self.buffers[target].append(observation)
+
+    def extend(self, dataset):
+        for x, y in dataset:  # TODO: This for loop sucks
```

Comment:
do we need to fix this before merging?

Reply:
idk, lets see if that slows down real training, maybe not